bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/values/BObject.java

```diff
@@ -36,7 +37,17 @@ public interface BObject extends RefValue {
 
     BFuture start(Strand strand, String funcName, Object... args);
 
-    Type getType();
+    /**
+     * Gets the type of ballerina object.
+     *
+     * @return Ballerina object type.
+     * @deprecated use {@link BObject#getOriginalType()} ()} instead.
+     * The API {@link BValue#getType()} should be used after fixing the cache issue.
+     */
+    @Deprecated
+    ObjectType getType();
+
+    Type getOriginalType();
```

Comment:
Shall we make default.
ex.     default ObjectType getOriginalType() {
        return getType();
    }

Reply:
So if some code implement this interface will not break.